gammapy/modeling/models/tests/test_temporal.py

```diff
@@ -47,76 +47,53 @@ def test_light_curve_evaluate(light_curve):
-    temporal_model.write(path=filename)
-    model_read = temporal_model.read(filename)
-    assert temporal_model.filename == filename
-    assert model_read.filename == filename
-    assert_allclose(model_read.table["TIME"].quantity.value, time.value)
-
-    t_ref = "2010-01-01T00:00:00"
+    time_ref = Time(55197.00000000, format='mjd')
+    livetime = 3.0 * u.hr
+    sigma = 0.5 * u.h
     t_min = "2010-01-01T00:00:00"
-    t_max = "2010-01-01T08:00:00"
-
-    sampler = temporal_model.sample_time(
-        n_events=2, t_min=t_min, t_max=t_max, random_state=0, t_delta="10 min"
+    t_max = "2010-01-01T03:00:00"
+    t_delta = "3 min"
+
+    times = time_ref + livetime * np.linspace(0, 1, 1000)
+    flare_model = GaussianTemporalModel(t_ref=(times[500].mjd) * u.d, sigma=sigma)
+
+    lc = Table()
+    meta = time_ref_to_dict(times[0])
+    lc.meta = meta
+    lc.meta["TIMEUNIT"] = 's'
+    lc["TIME"] = (times - times[0]).to("s")
+    lc["NORM"] = flare_model(times)
+
+    temporal_model = LightCurveTemplateTemporalModel(lc)
+    sampler_template = temporal_model.sample_time(
+        n_events=1000, t_min=t_min, t_max=t_max, random_state=0, t_delta=t_delta
     )
-    sampler = u.Quantity((sampler - Time(t_ref)).sec, "s")
-
-    assert len(sampler) == 2
-    assert_allclose(sampler.value, [12661.65802564, 7826.92991], rtol=1e-5)
+    assert len(sampler_template) == 1000
 
-    table = Table()
-    table["TIME"] = time
-    table["NORM"] = np.ones(len(time))
-    table.meta = dict(MJDREFI=55197.0, MJDREFF=0, TIMEUNIT="hour")
-    temporal_model_uniform = LightCurveTemplateTemporalModel(table)
-
-    sampler_uniform = temporal_model_uniform.sample_time(
-        n_events=2, t_min=t_min, t_max=t_max, random_state=0, t_delta="10 min"
-    )
-    sampler_uniform = u.Quantity((sampler_uniform - Time(t_ref)).sec, "s")
-
-    assert len(sampler_uniform) == 2
-    assert_allclose(sampler_uniform.value, [1261.65802564, 6026.9299098], rtol=1e-5)
-
-    temporal_model = ConstantTemporalModel()
-    sampler_costant = temporal_model.sample_time(
-        n_events=2, t_min=t_min, t_max=t_max, random_state=0
+    temporal_model = GaussianTemporalModel(t_ref=(time_ref.mjd) * u.d, sigma=sigma)
+    sampler_gauss = temporal_model.sample_time(
+        n_events=1000, t_min=t_min, t_max=t_max, random_state=0, t_delta=t_delta
     )
-    sampler_costant = u.Quantity((sampler_costant - Time(t_ref)).sec, "s")
-
-    assert len(sampler_costant) == 2
-    assert_allclose(sampler_costant.value, [4330.10377559, 3334.04566256], rtol=1e-5)
+    assert len(sampler_gauss) == 1000
 
-    temporal_model = ExpDecayTemporalModel(t_ref=Time(t_ref).mjd * u.d)
-    sampler_expo = temporal_model.sample_time(
-        n_events=2, t_min=t_min, t_max=t_max, random_state=0
-    )
-    sampler_expo = u.Quantity((sampler_expo.mjd - Time(t_ref).mjd), "d")
+    light = np.histogram(sampler_template.mjd, bins=30)
+    peak_time = light[1][np.argmax(light[0])]
```

Comment:
As the light curve is Gaussian here, maybe check both: the mean and standard deviation?